crypto/evp/evp_enc.c

```diff
@@ -471,6 +646,62 @@ int EVP_CipherUpdate(EVP_CIPHER_CTX *ctx, unsigned char *out, int *outl,
         return EVP_DecryptUpdate(ctx, out, outl, in, inl);
 }
 
+int EVP_CipherPipelineUpdate(EVP_CIPHER_CTX *ctx, size_t numpipes,
+                     unsigned char **out, int **outl,
+                     const unsigned char **in, int *inl)
```

Comment:
EVP APIs appear to take `int` and convert it to `size_t`, possibly for backward compatibility. As I am introducing new APIs, I wonder if I can use `size_t** outl` and `size_t *inl`.

Additionally, the use of a double pointer is confusing. The `**in` and `**out` signify arrays of pointers. But what does `**outl` represent? Would using an array of integers `*outl` be more efficient than an array of pointers for output length?

Reply:
Using `size_t` instead of `int` makes sense to me.

An array of integers also seems to make sense.

Reply:
just a bike-shed idea: seeing arguments `unsigned char **out, int **outl`, would it make sense to fold them to _iovec_-like structure?:
```
struct {
   unsigned char *buf;
   size_t buf_len;
} iovec_like;
```
and just pass array of structures? this would change prototype to something like:
```
int EVP_CipherPipelineUpdate(EVP_CIPHER_CTX *ctx, size_t numpipes,
                     struct iove_like *io_in,
                     struct iovec_like *io_out)
```
my take on `numpipe` is it says how many buffers/iovecs is in the array, is my assumption correct?

Reply:
The iovec style seems appropriate for this API. I will add iovec as an option in design document. Your understanding of `numpipes` is correct.